include/dr/shp/copy.hpp

```diff
@@ -75,19 +75,24 @@ template <std::forward_iterator InputIt, lib::distributed_iterator OutputIt>
   requires __detail::is_syclmemcopyable<std::iter_value_t<InputIt>,
                                         std::iter_value_t<OutputIt>>
 cl::sycl::event copy_async(InputIt first, InputIt last, OutputIt d_first) {
-  auto segments = lib::ranges::segments(d_first);
-  auto segment = std::begin(segments);
+  auto &&segments = lib::ranges::segments(d_first);
```

Comment:
`auto &&` is for forwarding only (+ some exceptions with good reasons) and should not be used for local variables. See [Why should one never use auto&& for local variables?](https://stackoverflow.com/questions/26355406/why-should-one-never-use-auto-for-local-variables).

Reply:
`segments()` returns a viewable range, which may include an lvalue reference.  I use `auto&&` here because it will bind `auto` to an lvalue reference, if one is returned, whereas `auto` will always bind to the value type.